internal/store/migrations/0026_work_thread.sql

```
@@ -0,0 +1,41 @@
+-- The thread FK is checked on every thread-row delete (the session cascade):
+-- indexed, the session_threads_parent_idx precedent.
+CREATE INDEX work_items_thread_idx ON work_items (thread_id) WHERE thread_id IS NOT NULL;
+
+-- Each thread declares its own MCP servers (its agent's mcp_servers), so a
+-- listing is per thread: the key widens the same way, NULL for the primary.
+-- A server two threads both declare is discovered once per thread.
+ALTER TABLE mcp_catalogs ADD COLUMN thread_id text REFERENCES session_threads(id) ON DELETE CASCADE;
+ALTER TABLE mcp_catalogs DROP CONSTRAINT mcp_catalogs_pkey;
+CREATE UNIQUE INDEX mcp_catalogs_thread_server_idx
+    ON mcp_catalogs (session_id, thread_id, server_name) NULLS NOT DISTINCT;
+CREATE INDEX mcp_catalogs_thread_idx ON mcp_catalogs (thread_id) WHERE thread_id IS NOT NULL;
```

SQLFluff (4.2.2)

[error] CREATE INDEX should use CONCURRENTLY to avoid locking the table during the build.

(PG01)